File: e182/python/solution.py
class Solution:
    """
    @param A: A positive integer which has N digits, A is a string.
    @param k: Remove k digits.
    @return: A string
    """
    def DeleteDigits(self, A, k):
        # write you code here
        n = len(A)
        ret_size = n - k
        ret = ''
        last_idx = -1
        for i in range(ret_size):
            minimal = 10
            for j in range(last_idx+1, i+k+1):
                tmp = int(A[j])
                if tmp < minimal:
                    minimal = tmp
                    last_idx = j
                    #print i,last_idx, minimal
            #print minimal
            ret += str(minimal)
        return ret.lstrip('0')

File: e182/java/Solution.java
public class Solution {
    /**
     *@param A: A positive integer which has N digits, A is a string.
     *@param k: Remove k digits.
     *@return: A string
     */
    public String DeleteDigits(String A, int k) {
        // write your code here
        StringBuilder stb = new StringBuilder();
        int lastIdx = -1;
        for (int i =0; i<A.length()-k; i++) {
            int min = 10;
            for (int j=lastIdx+1; j<=i+k; j++) {
                int tmp = A.charAt(j) - '0';
                if (tmp < min) {
                    min = tmp;
                    lastIdx = j;
                    
                }
            }
            stb.append(Integer.toString(min));
        }
        String ret = stb.toString();
        for (int l=0; l<ret.length(); l++) {
            if (ret.charAt(l) != '0') {
                ret = ret.substring(l);
                break;
            }
        }
        return ret;
        
    }
}
